5.drug-dependency/scripts/2.pearson-correlation.py

```diff
@@ -0,0 +1,195 @@
+        if latent_values.nunique() <= 1 or drug_values.nunique() <= 1:
+            corr = np.nan
+        else:
+            # Drop missing values for both columns
+            valid_data = pd.concat([latent_values, drug_values], axis=1).dropna()
+            latent_values_valid = valid_data[latent_col]
+            drug_values_valid = valid_data[drug_col]
+
+            if len(latent_values_valid) > 1 and len(drug_values_valid) > 1:
+                # Calculate Pearson correlation
+                corr, _ = pearsonr(latent_values_valid, drug_values_valid)
+            else:
+                corr = np.nan
+                print("nan")
+        
+        # Store the result
+        correlation_results.append({
+            'latent_dimension': latent_col,
+            'drug': drug_col,
+            'correlation': corr
+        })
+
+# Convert the results to a dataframe for easier analysis
+correlation_df = pd.DataFrame(correlation_results)
+
+# Display the correlation dataframe
+correlation_df.sort_values(by='correlation', key=abs, ascending=False).head(50)
+
+
+# In[17]:
+
+
+#Sort pathways by NES score (ascending order)
+pathway_df.sort_values(by='nes', ascending=True)
+
+ranked_gsea = pathway_df.sort_values(by='nes', key=abs, ascending=False)
+
+#Group by 'z_dim' and aggregate 'Term' into a list of associated pathways
+grouped_pathway_df = ranked_gsea.groupby('z_dim').apply(lambda x: x.nlargest(10, 'nes')['Term'].tolist()).reset_index(drop=False)
+
+# remove z_
+grouped_pathway_df['z_dim'] = grouped_pathway_df['z_dim'].str.replace('z_', '', regex=False)
+
+grouped_pathway_df.columns = ['latent dimension', 'Associated Pathways']
+
+grouped_pathway_df.head()
+
+
+# In[18]:
+
+
+# Assuming 'drug_column_name' is the column in prism_trt_df that matches the 'drug' column in correlation_df
+prism_trt_df_filtered = prism_trt_df[['column_name', 'name', 'moa', 'target']]
+
+# Merge correlation_df with prism_trt_df based on the 'drug' column in correlation_df and the matching column in prism_trt_df
+correlation_df_merged1 = pd.merge(correlation_df, prism_trt_df_filtered, how='left', left_on='drug', right_on='column_name')
+
+# Drop the redundant drug_column_name column after the merge if needed
+correlation_df_merged1 = correlation_df_merged1.drop(columns=['column_name'])
+
+# Merge correlation_df with prism_trt_df based on the 'drug' column in correlation_df and the matching column in prism_trt_df
+correlation_df_merged = pd.merge(correlation_df_merged1, grouped_pathway_df, how='left', left_on='latent_dimension', right_on='latent dimension')
+
+# Drop the redundant drug_column_name column after the merge if needed
+correlation_df_merged = correlation_df_merged.drop(columns=['latent dimension'])
+
+significant_corr_df = correlation_df_merged[
+    (correlation_df_merged['correlation'].abs() > 0.1)
+]
+# saving results as single output file
+correlation_dir = pathlib.Path("./results/drug_correlation.csv")
+significant_corr_df.to_csv(correlation_dir)
```

Comment:
I don't see that CSV in this PR. Recommend confirming that is this file created and is committed to the repository.